### ISSUE TITLE
Inconsistent results with enum.Enum's functional interface

### ISSUE THREAD

**Issue by jparise** [microsoft/pyright#1713]
`enum.Enum` supports [a functional interface](https://docs.python.org/3.9/library/enum.html#functional-api) for building `Enum` types:

> The second argument is the source of enumeration member names. It can be a whitespace-separated string of names, a sequence of names, a sequence of 2-tuples with key/value pairs, or a mapping (e.g. dictionary) of names to values.

Pyright correctly identifies `Pets` as a type (alias) when a whitespace-separated string of names is used:

```py
Pets = enum.Enum('Pets', 'DOG CAT')
```

... but if I pass a more complex type:

```py
Pets = enum.Enum('Pets', ('DOGS', 'CATS'))
```

... `Pets` is classified as a variable and therefore can't be used in contexts where a type is required.

I verified this with pyright 1.1.127.

**Comment by erictraut** [comment]
Yes, it's an intentional decision not to support all of the alternate variants of enum creation. The variant you're highlighting is typically used only in cases where the values are dynamically specified at runtime, and static type checkers can't deal with those anyway. If you are defining the enum names statically, please switch to the cleaner and more explicit class-based enum declaration:

```python
class Pets(Enum):
    Dogs = 0
    Cats = 1
```

As you've noted, pyright does support the variant where the list of names is passed as a single string literal.

**Comment by jparise** [comment]
That makes sense. My example above is simplified; I was a dynamic construction approach in the original code, and I see how that would be difficult for a type checker to understand.

I wonder if there would be a way to make this result more obvious, though. It's a little surprising that you can construct a (Python) type that can't be used as a type annotation. I don't have a specific suggestion, though. 😉

**Comment by erictraut** [comment]
The symbol can still be used as a type annotation, but it won't be of much use in terms of type checking. 

My recommendation is to avoid enums whose members are dynamically-defined. If you want dynamic behavior, use a more general container class, like a `dict`, that is designed for this.
